ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp

```diff
@@ -581,6 +579,17 @@ std::pair<ttnn::Tensor, std::optional<ttnn::Tensor>> prepare_conv_weights_biases
         weight_tensor_ = tt::tt_metal::convert_conv_weight_tensor_to_tiled_layout(
             weight_tensor_, weight_block_h_ntiles, weight_block_w_ntiles, weights_bias_dtype);
     }
+    //convert_conv_weight_tensor adds the padding to the base shape.
```

Comment:
Why does the function have to add the padding?

Please extend the sentence with a quick explanation.

Otherwise it seems like the right fix is to not add padding in the first place.